### PR TITLE
Add pressure sensor

### LINKED ISSUE #73: Interface with pressure sensor
1. Please describe your motivation for this feature. What benefits will this feature provide? What are the some examples of use cases for this feature?

    The pressure sensor will be able to provide information about the depth of the robot. This can then be used by sensor fusion to get a better estimate of our z-position.

2. Please describe the overall structure of your feature. What scripts/libraries do you plan on creating for your feature? How will your various scripts/libraries interact with each other? Will you need to modify any existing packages? Which workspace (i.e onboard, landside) will this feature live in?

    The Arduino code in the offboard_comms package will need to be modified to interface with the pressure sensor, using the library provided by Blue Robotics. A pressure to depth converter script will also be created and will be added to the data_pub package.

3. Give us some information about the input/output to your feature, if applicable. What specific topics will your feature subscribe/publish to? What services/actions will your feature call or contain? What message types will be used? What ROS parameters will you have? What packages is your feature dependent on?

    The Arduino code will publish the raw pressure data to `/offboard/pressure`. This will be of type `sensor_msgs/Fluid_Pressure`.
The pressure to depth converter will subscribe to `/offboard/pressure` and publish the converted depth data to `/sensors/depth`. `/sensors/depth` will be of type `std_msgs/Float64`.

4. Give us some information about your testing plan. How will you test your feature? Will you require pool testing, or can you simply test via simulation? How will you know that your feature works? 

    The code will first be tested on land by attaching the pressure sensor to an Arduino and starting both the Arduino and pressure to depth converter node. We will then submerge the pressure sensor in the sink or a bucket of water, and the test will be successful if the depth is accurate to ±5 cm. 
After this we will do test the pressure during pool testing by monitoring the depth as the robot submerges. We will measure note the measured depth at the bottom of the diving pool, which we know is 15 ft deep. We will also monitor the changes in the depth as the robot undergoes complex maneuvers in the rpy axes.
Finally, we will monitor the integration of the pressure sensor with sensor fusion, and will work with that team and the controls team to ensure that the state estimation still works and that controls will still be able to perform maneuvers and hold depth.

5. What do you need to be done by another team or sub-team for this feature to be fully integrated (e.g. hardware support from electronics or integration with motion planning)? Please either @mention the lead here, or contact them via Slack.

    Electronics - add pressure sensor to electronics stack
    CS/Sensor Fusion - add to state estimation
    Simulation - mock depth sensor

6. Finally, please @mention the people working with you on this feature, so that they will be linked to this issue.

### PR DESCRIPTION
**Pull request by muthuArivoli**
Add pressure sensor input and a converter to convert pressure to depth.  
Closes #73

### CODE REVIEW

**Comment by muthuArivoli** [comment]
To do before merge:
- [ ] Test whether thrusters/servos will still run with the updated mulitiplexed code
- [ ] Test whether the Arduino connects to the pressure sensor and publishes correct data
- [ ] Test whether the depth converter correctly converts pressures to depth

After this, maybe in a separate PR, someone will need to figure out how to integrate the depth readings into sensor fusion

File: onboard/catkin_ws/src/offboard_comms/Arduino Sketchbooks/ThrusterServoArduino/ThrusterServoArduino.ino [hunk 1]
```diff
@@ -58,6 +67,14 @@ void setup(){
         servos[i] = new MultiplexedServo(&pwm_multiplexer, i + NUM_THRUSTERS);
         servos[i]->initialise();
     }
+    
+    //Wire.begin() occurs in pwm_multiplexer.begin()
```
Comment (davidAmiron) [inline onboard/catkin_ws/src/offboard_comms/Arduino Sketchbooks/ThrusterServoArduino/ThrusterServoArduino.ino:71]: Some spaces (2?) between // and text

File: onboard/catkin_ws/src/offboard_comms/Arduino Sketchbooks/ThrusterServoArduino/ThrusterServoArduino.ino [hunk 2]
```diff
@@ -39,16 +42,22 @@ void servo_control_callback(const offboard_comms::SetServo::Request &sc_req, off
     sc_res.success = true;
 }
 
+//Message to use with the pressure sensor
```
Comment (davidAmiron) [inline onboard/catkin_ws/src/offboard_comms/Arduino Sketchbooks/ThrusterServoArduino/ThrusterServoArduino.ino:45]: Spaces

File: onboard/catkin_ws/src/offboard_comms/Arduino Sketchbooks/ThrusterServoArduino/ThrusterServoArduino.ino [hunk 3]
```diff
@@ -39,16 +42,22 @@ void servo_control_callback(const offboard_comms::SetServo::Request &sc_req, off
...
+sensor_msgs::FluidPressure pressure_msg;
+
 //Sets node handle to have 2 subscribers, 2 publishers, and 150 bytes for input and output buffer
```
Comment (davidAmiron) [inline onboard/catkin_ws/src/offboard_comms/Arduino Sketchbooks/ThrusterServoArduino/ThrusterServoArduino.ino:48]: Spaces

File: onboard/catkin_ws/src/offboard_comms/Arduino Sketchbooks/ThrusterServoArduino/MultiplexedObject.cpp [hunk 4]
```diff
@@ -2,31 +2,30 @@
...
     attached = false;
 }
 
 MultiplexedObject::~MultiplexedObject(){
-    detach(num);
+    detach();
 }
 
 void MultiplexedObject::attach(int _num){
     // make sure it isn't trying to run multiple pins
```
Comment (davidAmiron) [inline onboard/catkin_ws/src/offboard_comms/Arduino Sketchbooks/ThrusterServoArduino/MultiplexedObject.cpp:14]: Capital M

**Comment by muthuArivoli** [comment]
Tested this and it seems like there is not enough SRAM on the nano for this code. Will look into finding places to optimize memory usage, but it may simple be easier to switch to using a nano.

**Pull request merged**